### PR TITLE
Add seperate dashboard and submissiondetail views for partner and com…

### LINKED ISSUE #1147: Custom dashboard and submission detail views for reviewers, partners and community reviewers
### Dashboard

New custom dashboards for Partners and Community reviewers. Additions to reviewers dashboard.

**Reviewers:**

* Mainly write reviews and that part of the dashboard works well.
* They also can send in their own submissions and if they do a new section should display on their dashboard similar to what applicants see.

**Partners:**

* Should see the submission they have been added to as partners. They can both review and edit these submissions.
* Should see any reviews they write just like reviewers.
* Should see their own submissions if they send in any.

**Community reviewers:**

* Mainly write their own submissions as applicants.
* Should also see a list of any submissions that are in a COMMUNITY_REVIEW_PHASES state. These they can review.
* Should see any reviews they write just like reviewers.

### Submission detail view

New custom submission detail views for Partners and Community reviewers.

**Partners:**

* Everything like applicants
* See their own reviews and reviews set to be public.

**Community reviewers:**

* Everything like applicants
* See their own reviews and reviews set to be open to all reviewers.

### PR DESCRIPTION
**Pull request by frjo**
…munity reviewer roles.

Fixes #1147

Fixes https://github.com/OpenTechFund/issues.otf/issues/5

### CODE REVIEW

File: opentech/apply/dashboard/templates/dashboard/reviewer_dashboard.html [hunk 1]
```diff
@@ -36,6 +36,39 @@ <h1 class="gamma heading heading--no-margin heading--bold">Dashboard</h1>
 
         {% endif %}
     </div>
+
```
Comment (frjo) [inline opentech/apply/dashboard/templates/dashboard/reviewer_dashboard.html]: Maybe add a "if my_submissions" check to only show this if the reviewer has submissions, I believe most will not have any.

Comment (Parbhat) [reply opentech/apply/dashboard/templates/dashboard/reviewer_dashboard.html]: Thanks, updated.

File: opentech/apply/determinations/views.py [hunk 2]
```diff
@@ -335,5 +335,7 @@ def dispatch(self, request, *args, **kwargs):
 
 class DeterminationDetailView(ViewDispatcher):
     admin_view = AdminDeterminationDetailView
-    applicant_view = ApplicantDeterminationDetailView
     reviewer_view = ReviewerDeterminationDetailView
+    partner_view = ApplicantDeterminationDetailView
+    community_view = None
```
Comment (Parbhat) [inline opentech/apply/determinations/views.py]: @frjo Setting it to None will create a bug which will not allow community reviewers to see the determination for submissions submitted by them. What is the requirement for visibility of determination when the user is a community reviewer or a partner?

Comment (frjo) [reply opentech/apply/determinations/views.py]: @Parbhat community reviewers as well as partners can reuse the ReviewerDeterminationDetailView.

Everyone that can view a submission should also be able to view its determinations.

File: opentech/apply/funds/views.py [hunk 3]
```diff
@@ -466,23 +466,77 @@ class ReviewerSubmissionDetailView(ReviewContextMixin, ActivityContextMixin, Del
...
     form_views = [CommentFormView]
 
+    def dispatch(self, request, *args, **kwargs):
+        submission = self.get_object()
+        # If the requesting user submitted the application, return the Applicant view.
+        # Reviewers and partners may somtimes be appliants as well.
+        if submission.user == request.user:
+            return ApplicantSubmissionDetailView.as_view()(request, *args, **kwargs)
+        # Only allow reviewers in the submission they are added as reviewers or has reviewed earlier
+        reviewer_has_access = submission.reviewers.filter(pk=request.user.pk).exists() or submission.reviews.values('author').filter(author=request.user.pk)
```
Comment (frjo) [inline opentech/apply/funds/views.py]: New take on this from OTF. Reviewers should be able to view any submission that have reached an "AC review" state or higher and have not been dismissed.

Can we make use of "permissions.can_view()" for this? Or should we add something similar to COMMUNITY_REVIEW_PHASES where we list all the states reviewers should have access to, i.e. REVIEW_VIEW_ACCESS_PHASES? Other better solutions?

(Reviewers should, as it is now and have been for a long time, only be able to review submission they have been assigned to.)

Comment (frjo) [reply opentech/apply/funds/views.py]: If this needs some thinking we can revert to the current functionality where reviewers can see all submission. It has been like that from day one so no harm in postponing updating this a bit.

Comment (frjo) [reply opentech/apply/funds/views.py]: Getting the other nice improvements in this PR out is more important.

File: opentech/apply/determinations/views.py (hunk 2 above)
Comment (Parbhat) [reply opentech/apply/determinations/views.py]: @frjo Thanks for the information. We have to create new views for partners, community reviewers as these need different checks.

File: opentech/apply/funds/views.py (hunk 3 above)
Comment (Parbhat) [reply opentech/apply/funds/views.py]: Thanks for the information. We can use permissions.can_view() along with other required checks. It seems better to handle it in another PR. We also need to change `ReviewerDeterminationDetailView` as currently, a reviewer can see any determination as well.

**Comment by fourthletter** [comment]
The dashboard and submission detail view are visible for Partner and Community Reviewer during the test. Thanks!

I assigned this application https://test-apply.opentech.fund/apply/submissions/24/ to 1) Partner and 2) both "Test Reviewer" and "Maya Reviewer". The application appears in the Partner's dashboard, but NOT for community reviewers. 

image

**Comment by Parbhat** [comment]
@fourthletter I can not access the test URL but an application should be in the community review stage to appear in the dashboard of community reviewer.

**Pull request merged**